### DISCUSSION TITLE
Workflow wide file fast option

### DISCUSSION THREAD

**Discussion by nedrebo** [community/community#10221] [Actions] [answered]
I propose to add a workflow-wide `fail-fast` option. This option would be similar to `jobs.<job_id>.strategy.fail-fast` except it also works on free-standing jobs, not only "matrix" jobs.

When this feature is enabled all jobs should be sent the termination signal once any job fails. Then the workflow should not spawn new jobs but instead exit as quickly as possible. [1]

In cases where you for any reason have split your workflow in multiple pieces (multiple `.yml` files), I propose that fast-fail can also, based on an option, cancel all sibling workflows that were triggered by the same `on: push`. [2]

Both [1] and [2] are very important for us to not waste CI resources and reduce queues.

We currently use [`andymckay/cancel-action`](https://github.com/andymckay/cancel-action) to achieve [1], but it is inconvenient since we need to use a code generator to insert it into all jobs in all workflows. It also clutters the step view in the UI. A single top-level option would be much better.

We are now looking for a solution to [2] since `andymckay/cancel-action` does not support that at all.

**Comment by ethomson** [comment, marked as answer]
Thanks for the feedback - this is a good idea that we should implement.
